include/Hcal/HcalWABVetoProcessor.h

```diff
@@ -0,0 +1,63 @@
+/**
+ * @file HcalWABVetoProcessor.h
+ * @brief Processor that determines if an event is vetoed by the Hcal.
+ * @author Omar Moreno, SLAC National Accelerator Laboratory
+ */
+
+#ifndef __HCAL_HCAL_VETO_PROCESSOR_H__
+#define __HCAL_HCAL_VETO_PROCESSOR_H__
+
+//----------------//
+//   C++ StdLib   //
+//----------------//
+#include <string>
+
+//----------//
+//   LDMX   //
+//----------//
+#include "Event/HcalHit.h"
+#include "Ecal/Event/EcalHit.h"
+#include "Event/HcalCluster.h"
+#include "Event/HcalVetoResult.h"
+#include "Framework/Configure/Parameters.h"
+#include "Framework/EventProcessor.h"
+
+namespace hcal {
+
+class HcalWABVetoProcessor : public framework::Producer {
+ public:
+  /** Constructor */
+  HcalWABVetoProcessor(const std::string &name, framework::Process &process);
+
+  /** Destructor */
+  ~HcalWABVetoProcessor();
+
+  /**
+   * Configure the processor using the given user specified parameters.
+   *
+   * @param parameters Set of parameters used to configure this processor.
+   */
+  void configure(framework::config::Parameters &parameters) final override;
+
+  /**
+   * Run the processor and create a collection of results which
+   * indicate if the event passes/fails the Hcal veto.
+   *
+   * @param event The event to process.
+   */
+  void produce(framework::Event &event);
+
+ private:
+     double maxtotalEnergyCompare_{0.};
```

Comment:
Please document what each of these variables are used for as follows: 
`/// Example comment
double variable_{0.};
`

Reply:
> Do you have a talk or plots showing the performance of the veto? If so, can you link them to the PR for reference? Thanks!
> […](#)
> On Wed, Jan 5, 2022, 1:22 AM Sophie Middleton ***@***.***> wrote: First attempt at a WAB Veto - will require further work but seems fine for a first step. ------------------------------ You can view, comment on, or merge this pull request online at: #20 Commit Summary - [c34e701](https://github.com/LDMX-Software/Hcal/commit/c34e7011186593486481ce807969b9bb674cd9d3) <[c34e701](https://github.com/LDMX-Software/Hcal/pull/20/commits/c34e7011186593486481ce807969b9bb674cd9d3)> made veto, tested - [52e3ebc](https://github.com/LDMX-Software/Hcal/commit/52e3ebc78a70b1a2c7292802662ff2162139717d) <[52e3ebc](https://github.com/LDMX-Software/Hcal/pull/20/commits/52e3ebc78a70b1a2c7292802662ff2162139717d)> works veto File Changes (5 files <https://github.com/LDMX-Software/Hcal/pull/20/files>) - *M* include/Hcal/HcalClusterProducer.h <https://github.com/LDMX-Software/Hcal/pull/20/files#diff-416c44711346af73cb94f9048e26cd812385f5f65a87adbe121d1ed02aff1834> (3) - *A* include/Hcal/HcalWABVetoProcessor.h <https://github.com/LDMX-Software/Hcal/pull/20/files#diff-3d75a48792453a72c00c40fcb1b8736cfcf65a145a3558705e060d9c2763ef48> (63) - *M* python/hcal.py <https://github.com/LDMX-Software/Hcal/pull/20/files#diff-228a2321bd69304d346310d4d208ce2c24c81b4b2d3463b7787a4e222a0e3a28> (25) - *M* src/Hcal/HcalClusterProducer.cxx <https://github.com/LDMX-Software/Hcal/pull/20/files#diff-f5fbf15c068e96455c0f3b965836005ae9a0f652a8f5dc75c958c05cc66a60be> (4) - *A* src/Hcal/HcalWABVetoProcessor.cxx <https://github.com/LDMX-Software/Hcal/pull/20/files#diff-dd6f340ecce39cce8aac444a31c9beeece40dfa9602fb71e3c2b38a197078924> (103) Patch Links: - https://github.com/LDMX-Software/Hcal/pull/20.patch - https://github.com/LDMX-Software/Hcal/pull/20.diff — Reply to this email directly, view it on GitHub <#20>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AA4JMXBHRZTMPAVNWUQWSFTUUQEUVANCNFSM5LJNOMBQ> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Will do this, I am writing an Internal Note which will detail the Veto and all the WAB work. It's in the final stages and I can circulate it in the coming weeks. I'm at a conference this and next week but should be working inbetween talks to complete the document.

Reply:
I have done that

Reply:
Thank you!